Within robotpkg, a bare .so extention is always used, as this matches the
behaviour of python-cmake-modules>=0.12.

--- cmake/Modules/FindPythonExtra.cmake~	2022-03-01 20:48:08.000000000 +0100
+++ cmake/Modules/FindPythonExtra.cmake	2024-05-07 18:27:50.848036880 +0200
@@ -182,7 +182,7 @@
       "The SOABI suffix for Python native extensions. See PEP-3149: https://www.python.org/dev/peps/pep-3149/.")
   endif()
 
-  if(PYTHON_SOABI STREQUAL "" OR PYTHON_SOABI STREQUAL "None")
+  if(TRUE OR PYTHON_SOABI STREQUAL "" OR PYTHON_SOABI STREQUAL "None")
     set(PythonExtra_EXTENSION_SUFFIX
       ""
       CACHE INTERNAL
@@ -216,7 +216,7 @@
   PythonExtra_INCLUDE_DIRS
   PythonExtra_LIBRARIES
   PYTHON_SOABI)
-if(NOT WIN32)
+if(FALSE)
   list(APPEND _required_vars PythonExtra_EXTENSION_SUFFIX)
 elseif("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
   list(APPEND _required_vars PYTHON_EXECUTABLE_DEBUG)
